src/http_client_async.zig

```diff
@@ -780,8 +766,7 @@ pub const HTTPThread = struct {
 
     pub fn connect(this: *@This(), client: *HTTPClient, comptime is_ssl: bool) !NewHTTPContext(is_ssl).HTTPSocket {
         if (comptime is_ssl) {
-            const args = &client.custom_tls_props;
-            const needs_own_context = args.passphrase != null or args.ca != null or args.cert != null or args.key != null;
+            const needs_own_context = client.tls_props != null;
```

Comment:
What happens if we pass only`rejectUnauthorized` or `request_cert`? 
In this case request_cert should be ignored (always 1) and `rejectUnauthorized` should not trigger a custom context.